### PR TITLE
[NEAT-551] 🦭neat.verify Lookup in CDF

### PR DESCRIPTION
**Pull request by doctrino**
**Note** I think this solution can be improved if we move out the post validation outside `pydantic`. However, not sure about this, so that requires a discussion. 

## Without Client
image

## With Client
image

### CODE REVIEW

**Review by nikokaoja — APPROVED** [review]

File: cognite/neat/_rules/models/dms/_rules_input.py [hunk 1]
```diff
@@ -182,17 +191,25 @@ def _get_verified_cls(cls) -> type[DMSView]:
...
+        return output
+
+    def as_entity_id(self, default_space: str, default_version: str) -> ViewEntity:
+        return ViewEntity.load(self.view, strict=True, space=default_space, version=default_version)
+
+    def _load_implements(self, default_space: str, default_version: str) -> list[ViewEntity] | None:
+        return (
             [
-                ViewEntity.load(implement, space=default_space, version=default_version)
+                ViewEntity.load(implement, strict=True, space=default_space, version=default_version)
```
[the remark this answers is not in the document]
Comment (doctrino) [reply cognite/neat/_rules/models/dms/_rules_input.py:204]: Yes. I needed to have ContainerEntity and raise if it gave an unknown.

File: cognite/neat/_rules/models/dms/_rules_input.py [hunk 2]
```diff
@@ -287,3 +304,30 @@ def id_(self) -> URIRef:
         return DEFAULT_NAMESPACE[
             f"data-model/unverified/dms/{self.metadata.space}/{self.metadata.external_id}/{self.metadata.version}"
         ]
+
+    def referenced_views_and_containers(self) -> tuple[set[ViewEntity], set[ContainerEntity]]:
```
[the remark this answers is not in the document]
Comment (doctrino) [reply cognite/neat/_rules/models/dms/_rules_input.py:308]: Yes, they are referenced by the properties or views.

File: cognite/neat/_session/_base.py [hunk 3]
```diff
@@ -65,6 +66,28 @@ def version(self) -> str:
 
     def verify(self) -> IssueList:
         source_id, last_unverified_rule = self._state.data_model.last_unverified_rule
+
+        reference_rules: DMSInputRules | None = None
+        if isinstance(last_unverified_rule.rules, DMSInputRules):
+            dms_rules = last_unverified_rule.rules
+            views_ids, containers_ids = dms_rules.missing_views_and_containers_ids()
```
[the remark this answers is not in the document]
Comment (doctrino) [reply cognite/neat/_session/_base.py:73]: Good point I like imported.

File: cognite/neat/_session/_state.py [hunk 4]
```diff
@@ -146,3 +155,58 @@ def last_outcome(self) -> UploadResultList:
                 "No outcome available. Try using [bold].to.cdf.data_model[/bold] to upload a data model."
             )
         return self.outcome[-1]
+
+    def lookup_containers(
+        self, client: CogniteClient, container_ids: Sequence[dm.ContainerId]
+    ) -> list[dm.ContainerApply]:
+        if missing := set(container_ids) - set(self._cdf_containers.keys()):
```
[the remark this answers is not in the document]
Comment (doctrino) [reply cognite/neat/_session/_state.py:162]: It is caching to avoid looking up the same containers more than once.

**Pull request merged**